torchchat/generate.py

```diff
@@ -553,12 +552,10 @@ def decode_n_tokens(
                     **sampling_kwargs,
                 )
                 input_pos += 1
-                new_tokens.append(next_token.clone())
-                callback(new_tokens[-1], done_generating=_i == num_new_tokens - 2)
-                if need_probs or next_prob is None:
+                callback(next_token.clone(), done_generating=_i == num_new_tokens - 2)
+                if not need_probs or next_prob is None:
```

Comment:
Everything else is just cleaning up unused code

`not need_probs` is the only real change and in the non-speculative path is always false so the old check is effectively just `if next_prob is None`
https://github.com/pytorch/torchchat/blob/701d82658d0a48799131a7f6692574648068f118/torchchat/generate.py#L799

I think we should drop the check instead of negating here
```suggestion
                if next_prob is None:
```

Reply:
Honestly this is a nit we can  just merge